visearch-android/src/main/java/com/visenze/visearch/android/ProductSearch.java

```diff
@@ -0,0 +1,140 @@
+package com.visenze.visearch.android;
+
+import android.content.Context;
+
+import com.visenze.datatracking.SessionManager;
+import com.visenze.datatracking.VisenzeAnalytics;
+import com.visenze.datatracking.data.DataCollection;
+import com.visenze.visearch.android.model.ProductResponse;
+import com.visenze.visearch.android.network.ProductSearchService;
+
+import java.net.URL;
+
+public class ProductSearch {
+    private static final String USER_AGENT = "productsearch-android-sdk";
+    //    private static final String SEARCH_URL = "https://search-dev.visenze.com/v1/";
+    private static final String SEARCH_URL = "https://search.visenze.com/v1/";
+
+
+    private String uid;
+    private VisenzeAnalytics visenzeAnalytics;
+    private ProductSearchService productSearchService;
+
+    private ProductSearch(Context context, String appKey, String placementId, String endPoint, String userAgent, String uid) {
```

Comment:
placementId is Integer

Reply:
we also need another constructor without endpoint, userAgent, uid. These should be set automatically if not provided

Reply:
oh never mind, i just notice this is private

Reply:
changed already